lib/data/entity.js

```diff
@@ -417,6 +475,26 @@ const getWithConflictDetails = (defs, audits, relevantToConflict) => {
 
   const relevantBaseVersions = new Set();
 
+  // build up branches
+  const branches = new Map();
+  for (const version of defs) {
```

Comment:
This ends up looping over `defs` twice, which I don't think we need to do. I think we can build up `result` and `branches` in the same loop, as no version/def needs to look ahead to a later version, including for branch information.